lib/assets/chart_builder/main.js

```diff
@@ -0,0 +1,289 @@
+            <div class="field">
+              <label class="input-label">Type</label>
+              <select class="input input--sm" name="color_type">
+                <option value=""></option>
+                <option value="quantitative">Quantitative</option>
+                <option value="nominal">Nominal</option>
+                <option value="ordinal">Ordinal</option>
+              </select>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `;
+
+  const channelFields = ["x_axis", "y_axis", "color"];
+  const typeFields = ["x_axis_type", "y_axis_type", "color_type"];
+
+  updateInfoBox(info.missing_dep);
+  info.fresh ? setFreshData(info.options) : renderCurrentData(info.fields);
+
+  ctx.root.addEventListener("blur", handleFieldChange, true);
+  ctx.root.addEventListener("change", handleFieldChange);
+
+  function handleFieldChange(event) {
+    const { name, value } = event.target;
+    if (channelFields.includes(name)) { maybeReset({name, value}) };
+    ctx.pushEvent("update_field", { field: name, value });
+  }
+
+  ctx.handleEvent("update", ({ fields }) => {
+    clearUnavailableData();
+    setValues(fields);
+  });
+
+  ctx.handleEvent("missing_dep", ({ dep }) => {
+    updateInfoBox(dep);
+  });
+
+  ctx.handleEvent("set_axis_options", ({ options }) => {
+    options ? setAxisOptions(options) : missingData();
+  });
+
+  ctx.handleEvent("set_available_data", ({ options }) => {
+    setAvailableData(options);
+  });
+
+  function setValues(fields) {
+    for (const field in fields) {
+      ctx.root.querySelector(`[name="${field}"]`).value = fields[field];
+    }
+  }
+
+  function setAvailableData(options) {
+    enableAll();
+    const selectElem = ctx.root.querySelector(`[name="data"]`);
+    const selectedData = selectElem.value;
+    const availableData = options[selectedData];
+    selectElem.innerHTML = "";
+    createOptions(selectElem, Object.keys(options));
+    selectElem.value = selectedData;
+    if (availableData) {
+      setAvailableDataOptions(availableData);
+      selectElem.classList.remove("unavailable");
+    } else {
+      createOptions(selectElem, [selectedData], "unavailable-option");
+      selectElem.value = selectedData;
+      disableAll();
+      selectElem.disabled = false;
+      selectElem.classList.add("unavailable");
+    }
+  }
+
+  function renderCurrentData(fields) {
+    disableAll();
+    for (const field of channelFields.concat(["data"])) {
+      const elem = ctx.root.querySelector(`[name="${field}"]`);
+      elem.innerHTML = `<option value="${fields[field]}">${fields[field]}</option>`;
+    }
+    setValues(fields);
+    updateDataInfoBox(fields["data"]);
+  }
+
+  function setFreshData(options) {
+    enableAll();
+    const selectElem = ctx.root.querySelector(`[name="data"]`);
+    selectElem.innerHTML = "";
+    createOptions(selectElem, Object.keys(options));
+    const selected = selectElem.value;
+    updateDataInfoBox(options);
```

Comment:
The data info message no longer shows, this should probably be

```suggestion
    updateDataInfoBox(options[selected]);
```